packages/wdio-cli/src/launcher.js

```diff
@@ -99,7 +99,7 @@ class Launcher {
         if (this.isMultiremote) {
             return new Promise((resolve) => {
                 this.resolve = resolve
-                this.startInstance(this.configParser.getSpecs(), caps, 0)
+                this.startInstance({ files: this.configParser.getSpecs() }, caps, 0)
```

Comment:
What is the advantage of passing in an object instead of an array?

Reply:
The list of specs to run per capability is stored in `this.schedule[cid]`. Previously, there were no retries, and runner ids were generated sequentially. Now, each list of specs could possibly be accompanied by a retry count as well as a runner id override (to ensure the retry for '0-0' runs as '0-0' again). Changing `this.schedule[cid] = specs` to `this.schedule[cid] = { files: specs, cid: ..., retries: ... }` allows tracking of the additional fields nicely.

What we could do is to make `rid` and `retries` separate parameters to `startInstance` and keep `specs` as an array. Inside `startInstance`, the line `cid = this.getRunnerId(cid)` should then be changed to `cid = rid || this.getRunnerId(cid)`. Something along those lines. I do agree that is cleaner.